src/backend/distributed/deparser/deparse_view_stmts.c

```diff
@@ -92,3 +99,206 @@ AppendViewNameList(StringInfo buf, List *viewNamesList)
 		isFirstView = false;
 	}
 }
+
+
+/*
+ * DeparseAlterViewStmt deparses the given ALTER VIEW statement.
+ */
+char *
+DeparseAlterViewStmt(Node *node)
+{
+	AlterTableStmt *stmt = castNode(AlterTableStmt, node);
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+	AppendAlterViewStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+static void
+AppendAlterViewStmt(StringInfo buf, AlterTableStmt *stmt)
+{
+	const char *identifier = quote_qualified_identifier(stmt->relation->schemaname,
+														stmt->relation->relname);
+
+	appendStringInfo(buf, "ALTER VIEW %s ", identifier);
+
+	AlterTableCmd *alterTableCmd = castNode(AlterTableCmd, lfirst(list_head(stmt->cmds)));
+	AppendAlterViewCmd(buf, alterTableCmd);
+
+	appendStringInfoString(buf, ";");
+}
+
+
+static void
+AppendAlterViewCmd(StringInfo buf, AlterTableCmd *alterTableCmd)
+{
+	switch (alterTableCmd->subtype)
+	{
+		case AT_ChangeOwner:
+		{
+			AppendAlterViewOwnerStmt(buf, alterTableCmd);
+			break;
+		}
+
+		case AT_SetRelOptions:
+		{
+			AppendAlterViewSetOptionsStmt(buf, alterTableCmd);
+			break;
+		}
+
+		case AT_ResetRelOptions:
+		{
+			AppendAlterViewResetOptionsStmt(buf, alterTableCmd);
+			break;
+		}
+
+		case AT_ColumnDefault:
+		{
+			elog(ERROR, "Citus doesn't support setting or resetting default values for a "
+						"column of view");
+			break;
+		}
+
+		default:
+		{
+			ereport(ERROR, (errmsg("unsupported subtype for alter view command"),
+							errdetail("sub command type: %d", alterTableCmd->subtype)));
+		}
+	}
+}
+
+
+static void
+AppendAlterViewOwnerStmt(StringInfo buf, AlterTableCmd *alterTableCmd)
+{
+	appendStringInfo(buf, "OWNER TO %s", RoleSpecString(alterTableCmd->newowner, true));
+}
+
+
+static void
+AppendAlterViewSetOptionsStmt(StringInfo buf, AlterTableCmd *alterTableCmd)
+{
+	ListCell *lc = NULL;
+	bool initialOption = true;
+	foreach(lc, (List *) alterTableCmd->def)
+	{
+		DefElem *def = (DefElem *) lfirst(lc);
+
+		if (initialOption)
+		{
+			appendStringInfo(buf, "SET (");
+			initialOption = false;
+		}
+		else
+		{
+			appendStringInfo(buf, ",");
+		}
+
+		appendStringInfo(buf, "%s", def->defname);
+		if (def->arg != NULL)
+		{
+			appendStringInfo(buf, "=");
+			appendStringInfo(buf, "%s", defGetString(def));
+		}
+	}
+
+	appendStringInfo(buf, ")");
```

Comment:
do we need `;` at the end? Just in case we send multiple commands in one round-trip for some reason in the future. Maybe go over the commands and add if needed?

Reply:
We add `;` via `AppendAlterViewStmt`. I've also checked other commands.